### DISCUSSION TITLE
Failed building wheel for httptools

### DISCUSSION THREAD

**Discussion by zero0nee** [fastapi/fastapi#8198] [Questions]
After trying `pip3 install uvicorn` i get the error message `Failed building wheel for httptools`.

I know this is a duplication of this error (https://github.com/MagicStack/httptools/issues/33), but I can't seem to find a solution. Anyone having the same problem, or know of a solution? 

I'm on macOS Mojave 0.14.2
with python 3.7.1, but on 3.6.3 I had the same issue.

**Comment by zero0nee** [comment]
Updating to the latest version of Xcode, under "Software Updates" in "System Preferences" solved it for me on Mac.

**Comment by varunchopra** [comment]
For those facing this on Debian: `apt install python3-dev build-essential` is required.

**Comment by tiangolo** [comment]
Thanks for repoting back @zero0nee and thanks for the tip @chopraaa !

**Comment by h1code2** [comment]
> For those facing this on Debian: `apt install python3-dev build-essential` is required.
Thank you very much, your answer is very useful.
